src/oc_reporter_service.erl

```
@@ -0,0 +1,175 @@
+%%%------------------------------------------------------------------------
+%% Copyright 2019, OpenCensus Authors
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc Exports spans to Opencensus agent
+%% @end
+%%%------------------------------------------------------------------------
+
+-module(oc_reporter_service).
+
+-behaviour(oc_reporter).
+
+-include_lib("opencensus/include/opencensus.hrl").
+
+-export([init/1,
+         report/2]).
+
+-export([to_proto_map/1]).
+
+init(_Opts) ->
+    %% in case this is called before the app has booted
+    application:ensure_all_started(opencensus_service).
```

Shouldn't we check on return value here?

---

Probably not necessary. It'll be started by the release and fail there and be handled.

---

Then I would not try to launch it again in `init/1`. instead just let it fail if someone didn't configured it properly.

---

Eh, why make it more complicated and potentially confusing. If there was a builtin way to set these sorts of dynamic dependencies I'd agree.